sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/drmemory/CxxDrMemorySensor.java

```diff
@@ -0,0 +1,112 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.cxx.drmemory;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.sonar.api.batch.SensorContext;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.component.ResourcePerspectives;
+import org.sonar.api.config.Settings;
+import org.sonar.api.profiles.RulesProfile;
+import org.sonar.api.resources.Project;
+import org.sonar.plugins.cxx.drmemory.DrMemoryParser.DrMemoryError;
+import org.sonar.plugins.cxx.drmemory.DrMemoryParser.DrMemoryError.Location;
+import org.sonar.plugins.cxx.utils.CxxMetrics;
+import org.sonar.plugins.cxx.utils.CxxReportSensor;
+import org.sonar.plugins.cxx.utils.CxxUtils;
+import org.sonar.plugins.cxx.utils.EmptyReportException;
+
+/**
+ * Dr. Memory is a memory monitoring tool capable of identifying memory-related
+ * programming errors such as accesses of uninitialized memory, accesses to
+ * unaddressable memory (including outside of allocated heap units and heap
+ * underflow and overflow), accesses to freed memory, double frees, memory
+ * leaks, and (on Windows) handle leaks, GDI API usage errors, and accesses to
+ * un-reserved thread local storage slots. See also: http://drmemory.org
+ *
+ * @author asylvestre
+ */
+public class CxxDrMemorySensor extends CxxReportSensor {
+
+	public static final String REPORT_PATH_KEY = "sonar.cxx.drmemory.reportPath";
+	private final RulesProfile profile;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public CxxDrMemorySensor(ResourcePerspectives perspectives,
+			Settings settings, FileSystem fs, RulesProfile profile) {
+		super(perspectives, settings, fs, CxxMetrics.DRMEMORY);
+		this.profile = profile;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean shouldExecuteOnProject(Project project) {
+		return super.shouldExecuteOnProject(project)
+				&& !profile.getActiveRulesByRepository(
+						CxxDrMemoryRuleRepository.KEY).isEmpty();
+	}
+
+	@Override
+	protected String reportPathKey() {
+		return REPORT_PATH_KEY;
+	}
+
+	@Override
+	protected void processReport(final Project project,
+			final SensorContext context, File report)
+			throws javax.xml.stream.XMLStreamException {
+		CxxUtils.LOG.debug("Parsing 'Dr Memory' format");
+
+		try {
+			for (DrMemoryError error : DrMemoryParser.parse(report)) {
+				if (error.stackTrace.isEmpty()) {
+					saveUniqueViolation(project, context, CxxDrMemoryRuleRepository.KEY,
+							null, null,
+							error.type.getId(), error.message);
+				}
+				for (Location errorLocation : error.stackTrace) {
+					if (isFileInAnalysis(errorLocation)) {
+						saveUniqueViolation(project, context, CxxDrMemoryRuleRepository.KEY,
+								errorLocation.file,	errorLocation.line.toString(),
+								error.type.getId(), error.message);
+						break;
+					}
+
+				}
+			}
+		} catch (IOException e1) {
+			throw new EmptyReportException();
```

Comment:
@arnaudsylvestre please have a look to https://github.com/SonarOpenCommunity/sonar-cxx/blob/master/sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxReportSensor.java. EmptyReportException  is a special case.

```Java
  public void analyse(Project project, SensorContext context) {
    try {
      List<File> reports = getReports(settings, fs.baseDir(), reportPathKey());
      violationsCount = 0;
      
      for (File report : reports) {
        int prevViolationsCount = violationsCount;
        CxxUtils.LOG.info("Processing report '{}'", report);
        try {
          processReport(project, context, report);
          CxxUtils.LOG.debug("{} processed = {}", metric == null ? "Issues" : metric.getName(),
             violationsCount - prevViolationsCount);
        } catch (EmptyReportException e) {
          CxxUtils.LOG.warn("The report '{}' seems to be empty, ignoring.", report);
        }
      }

      CxxUtils.LOG.info("{} processed = {}", metric == null ? "Issues" : metric.getName(),
        violationsCount);
          
      if (metric != null) {
        Measure measure = new Measure(metric);
        measure.setIntValue(violationsCount);
        context.saveMeasure(measure);
      }
    } catch (Exception e) {
      String msg = new StringBuilder()
        .append("Cannot feed the data into sonar, details: '")
        .append(e)
        .append("'")
        .toString();
      throw new IllegalStateException(msg, e);
    }
  }
```